src/render_tasks/d3d12_deferred_composition.hpp

```diff
@@ -34,16 +34,23 @@ namespace wr
 
 			d3d12::desc::DescriptorHeapDesc heap_desc;
 			heap_desc.m_shader_visible = true;
-			heap_desc.m_num_descriptors = 3;
+			heap_desc.m_num_descriptors = 4;
 			heap_desc.m_type = DescriptorHeapType::DESC_HEAP_TYPE_CBV_SRV_UAV;
+			heap_desc.m_versions = d3d12::settings::num_back_buffers;
 			data.out_srv_heap = d3d12::CreateDescriptorHeap(n_render_system.m_device, heap_desc);
 
-			auto cpu_handle = d3d12::GetCPUHandle(data.out_srv_heap);
+			for (uint32_t i = 0; i < d3d12::settings::num_back_buffers; ++i)
+			{
+
+				auto cpu_handle = d3d12::GetCPUHandle(data.out_srv_heap, i);
+
+				auto deferred_main_data = fg->GetData<DeferredMainTaskData>();
+				auto deferred_main_rt = data.out_deferred_main_rt = static_cast<D3D12RenderTarget*>(deferred_main_data.m_render_target);
+				d3d12::CreateSRVFromRTV(deferred_main_rt, cpu_handle, 2, deferred_main_data.m_rt_properties.m_rtv_formats.data());
+				d3d12::CreateSRVFromDSV(deferred_main_rt, cpu_handle);
+				d3d12::CreateSRVFromStructuredBuffer(n_render_system.GetLightBuffer(), cpu_handle, i);
```

Comment:
The scene graph should return the light buffer. The render system has no idea what scene graph is being rendered. We want to be able to switch scene graphs.

Reply:
Not available yet - but, Koen is now working on a StructuredBuffer resource pool and it will be commited into this branch